### PR TITLE
A few miscellaneous comment fixes and a tiny code clarification.

### PR DESCRIPTION
**Pull request by benjaminp**

### CODE REVIEW

File: compiler/rustc_infer/src/infer/higher_ranked/mod.rs [hunk 1]
```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         // as-is, we need to do some extra work here in order to make sure
         // that function subtyping works correctly with respect to regions
         //
-        // Note: this is a subtle algorithm.  For a full explanation,
-        // please see the large comment at the end of the file in the (inlined) module
-        // `doc`.
+        // Note: this is a subtle algorithm.  For a full explanation, please see
+        // the rustc book:
```
Comment (jonas-schievink) [inline compiler/rustc_infer/src/infer/higher_ranked/mod.rs:28]:
```suggestion
        // the rustc development guide:
```

The [rustc book](https://doc.rust-lang.org/rustc/) is a different book

**Comment by crlf0710** [comment]
Ping from triage: @benjaminp Hi, would you mind rebasing the changes?

File: compiler/rustc_typeck/src/check/method/probe.rs [hunk 2]
```diff
@@ -765,12 +765,11 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
...
         });
 
-        // It is illegal to invoke a method on a trait instance that
-        // refers to the `Self` type. An error will be reported by
-        // `enforce_object_limitations()` if the method refers to the
-        // `Self` type anywhere other than the receiver. Here, we use
-        // a substitution that replaces `Self` with the object type
-        // itself. Hence, a `&self` method will wind up with an
+        // It is illegal to invoke a method on a trait instance that refers to
+        // the `Self` type. An error will be reported by `object_safety.rs` if
```
Comment (jyn514) [inline compiler/rustc_typeck/src/check/method/probe.rs:769]: Wow, I had to do some archeology for this! The comment was first introduced in 1bceb98084b0e1010d123f48785a89bc7ad7e40f all the way back in 2013. `fn enforce_object_limitations` was removed in 8d8d8d4e5292c2fa4a622d981a5f85fd3d8f34d0 and replaced by `fn check_object_safety`, but the comment was never updated. Then `check_object_safety` gets removed in 843db01bd925279da0a56efde532c9e3ecf73610 and the check is moved to `report_selection_error` in src/librustc/middle/traits/error_reporting.rs. That got me far enough that I found that the error is `SupertraitSelf` and now lives in `compiler/rustc_trait_selection/src/traits/object_safety.rs`.

In short I think it would be helpful to mention `SupertraitSelf` so other people can find this in the future.
```suggestion
        // the `Self` type. [`ObjectSafetyViolation::SupertraitSelf`] will be reported by `object_safety.rs` if
```

File: compiler/rustc_infer/src/infer/higher_ranked/mod.rs (hunk 1 above)
Comment (jyn514) [inline compiler/rustc_infer/src/infer/higher_ranked/mod.rs]: nit: can you squash https://github.com/rust-lang/rust/pull/78177/commits/9d49d63300bbea5fbec89e9e36be4b88ddb3336d into https://github.com/rust-lang/rust/pull/78177/commits/185ce4636c4f194b8ecb22e5e16e1c38004067e7 ?

File: compiler/rustc_infer/src/infer/higher_ranked/mod.rs [hunk 3]
```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
...
+        // https://rustc-dev-guide.rust-lang.org/traits/hrtb.html
```
Comment (jyn514) [inline compiler/rustc_infer/src/infer/higher_ranked/mod.rs:29]: I think https://rustc-dev-guide.rust-lang.org/borrow_check/region_inference/placeholders_and_universes.html might be a better link, it's the one shown in https://rustc-dev-guide.rust-lang.org/appendix/glossary.html?highlight=skolem#glossary for 'skolemization', which is what the comment originally said when introduced in 23652efffb32de3f2babcb6204f49268b19cc0e6.
```suggestion
        // <https://rustc-dev-guide.rust-lang.org/borrow_check/region_inference/placeholders_and_universes.html>
```

**Review by jyn514 — COMMENTED** [review]
This was fun to review :smile:

File: compiler/rustc_infer/src/infer/higher_ranked/mod.rs [hunk 4]
```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         // as-is, we need to do some extra work here in order to make sure
         // that function subtyping works correctly with respect to regions
         //
-        // Note: this is a subtle algorithm.  For a full explanation,
-        // please see the large comment at the end of the file in the (inlined) module
-        // `doc`.
+        // Note: this is a subtle algorithm.  For a full explanation, please see
+        // the rustc development guide:
```
Comment (camelid) [inline compiler/rustc_infer/src/infer/higher_ranked/mod.rs:28]: It's not a big deal, but people almost always refer to it as the 'rustc-dev-guide' or 'rustc dev guide'. So it's possible that it may confuse people to use a longer name. But, like I said it's not a big deal :)

Comment (camelid) [reply compiler/rustc_infer/src/infer/higher_ranked/mod.rs]: Wow, you're fast :)

**Comment by jyn514** [comment]
@bors r+ rollup

Great catch on all of these!
